src/main/java/com/example/provider/json/TripSerializer.java

```diff
@@ -69,9 +71,20 @@ public JsonElement serialize(Trip src, Type typeOfSrc, JsonSerializationContext
             .add(dropoffWaypoint)
             .build();
 
+    List<SerializedLatLng> serializedRouteList = new ArrayList<SerializedLatLng>();
```

Comment:
Nit: consider a 'Functional programming' approach here by mapping the list (We do something similar right above with 'intermediateDestinations'

```java
List<SerializedLatLng> serializedRouteList = src.getRouteList().stream().map(route -> 
  SerializedLatLng.newBuilder()
                .setLatitude(route.getLatitude())
                .setLongitude(route.getLongitude())
                .build()).collect(Collectors.toList());
```